search/includes/classes/class-search.php

```diff
@@ -688,6 +690,77 @@ public function filter__ep_bulk_items_per_page() {
 		return 500;
 	}
 
+	/**
+	 * Filter to return value of option that caches index_exists request (if it doesn't exist, cache it).
+	 * 
+	 * @param  array  $request  New remote request response
+	 * @param  array  $query    Remote request arguments
+	 * @param  array  $args     Request arguments
+	 * @param  array  $failures Number of failures
+	 * @param  string $type     Type of request
+	 * @return array  $request  New request
+	 */
+	public function get_cached_index_exists_request( $request, $query, $args, $failures = 0, $type = null ) {
+		if ( 'index_exists' !== $type ) {
+			return $request;
+		}
+
+		$option_name = $this->get_index_exists_option_name( $query['url'] );
+		$request     = get_option( $option_name );
+		if ( false === $request ) {
+			// Option doesn't exist, do the request and cache it.
+			$request = vip_safe_wp_remote_request( $query['url'] );
+			if ( ! is_wp_error( $request ) ) {
+				update_option( $option_name, $request );
+			}
+		}
+		
+		return $request;
+	}
+
+	/**
+	 * Invalidate cached index_exists request option on certain index actions.
+	 * 
+	 * @param  array  $request  New remote request response
+	 * @param  array  $query    Remote request arguments
+	 * @param  array  $args     Request arguments
+	 * @param  array  $failures Number of failures
+	 * @param  string $type     Type of request
+	 * @return array  $request  New request
+	 */
+	public function invalidate_cached_index_exists_request( $request, $query, $args, $failures = 0, $type = null ) {
+		$index_actions = [
+			'delete_index',
+			'refresh_indices',
+			'put_mapping',
+			'bulk_index',
+		];
+		if ( ! in_array( $type, $index_actions, true ) ) {
+			return $request;
+		}
+
+		$option_name = $this->get_index_exists_option_name( $query['url'] );
+		$option      = get_option( $option_name );
+		if ( false !== $option ) {
+			delete_option( $option_name );
+		}
```

Comment:
Minor nitpick, but can just `delete_option()` option here directly without checking get_option first. WP won't delete if it can't SELECT first an ID to delete anyway.